packages/xo-web/src/xo-app/disk-import/index.js

```diff
@@ -0,0 +1,193 @@
+import * as FormGrid from 'form-grid'
+import _, { messages } from 'intl'
+import ActionButton from 'action-button'
+import Button from 'button'
+import Collapse from 'collapse'
+import decorate from 'apply-decorators'
+import Dropzone from 'dropzone'
+import fromEvent from 'promise-toolbox/fromEvent'
+import React, { Component } from 'react'
+import { formatSize, mapPlus } from 'utils'
+import { generateId, linkState } from 'reaclette-utils'
+import { importDisks } from 'xo'
+import { injectIntl } from 'react-intl'
+import { injectState, provideState } from 'reaclette'
+import { readCapacityAndGrainTable } from 'xo-vmdk-to-vhd'
+import { SelectSr } from 'select-objects'
+
+const getInitialState = () => ({
+  disks: [],
+  mapDescriptions: {},
+  mapNames: {},
+  sr: undefined,
+})
+
+const DiskImport = decorate([
+  provideState({
+    initialState: getInitialState,
+    effects: {
+      handleDrop: (effects, files) => async ({ sr }) => {
+        const disks = []
+        await Promise.all(
+          mapPlus(files, async (file, push) => {
```

Comment:
`push` is never used and if you're using an `async` function, I guess that you want `Promise.all` to `await` all the promises, which it won't do because nothing is ever `push`ed.